code/Controllers/CMSMain.php

```diff
@@ -491,13 +491,11 @@ public function LinkPreview()
      */
     public function SiteTreeAsUL()
     {
-        // Pre-cache sitetree version numbers for querying efficiency
-        Versioned::prepopulate_versionnumber_cache(SiteTree::class, Versioned::DRAFT);
-        Versioned::prepopulate_versionnumber_cache(SiteTree::class, Versioned::LIVE);
-
-        if (method_exists(Hierarchy::class, 'prepopulate_numchildren_cache')) {
-            Hierarchy::prepopulate_numchildren_cache(SiteTree::class, Versioned::DRAFT);
-        }
+        $filter = $this->getSearchFilter();
+        SiteTree::singleton()->prepopulateTreeDataCache(null, [
```

Comment:
I think we should be consistent here and use `$this->config()->get('tree_class')` to decide what we are caching.

Reply:
Yes that would make sense. singleton($this->config()->get('tree_class'))`` most likely?